### DISCUSSION TITLE
should tag in snapshot's default .Version have a v prefix ?

### DISCUSSION THREAD

**Discussion by doodlesbykumbi** [goreleaser/goreleaser#2114] [Q&A] [answered]
`.Version` for a release has the v prefix stripped. I would expect the same for the snapshot. From https://goreleaser.com/customization/templates/:

> the version being released (v prefix stripped),
> or {{ .Tag }}-SNAPSHOT-{{ .ShortCommit }} in case of snapshot release

At present, any other templates where I use `.Version` but need to re-add the v prefix (because it might be relevant in a URL for example) ends up having a double prefix e.g. `vv0.5.4-SNAPSHOT-0b768c4`.

I've got a patch to fix this https://github.com/goreleaser/goreleaser/compare/master...doodlesbykumbi:patch-1 but wanted to get some input before submitting it.

**Comment by caarlos0** [comment, marked as answer]
You can customize your snapshot name template and use either Version or Tag depending on what you need... I don't think we need to change anything in goreleaser for that 🤔

**Reply by doodlesbykumbi** [reply]
@caarlos0 You can customize the template but the funcs available inside the template package are limited. My patch does the following
1. Add `trimprefix` func to the template package
2. Use `trimprefix` in the default so that it becomes `{{ trimprefix .Tag "v" }}-SNAPSHOT-{{ .ShortCommit }}`

**Reply by caarlos0** [reply]
I can agree with adding the trimprefix function, but not in changing the default as it might break other people...

**Reply by doodlesbykumbi** [reply]
Fair. I'm a bit on the fence though... On one end I would prefer to have consistent default behavior, on the other there really might be someone relying on this behavior though I would hazard a guess that it's highly unlikely. Either way I'll whip up a PR to add trimprefix

**Reply by doodlesbykumbi** [reply]
There we go https://github.com/goreleaser/goreleaser/pull/2116

**Reply by caarlos0** [reply]
merged, thanks @doodlesbykumbi 🙏
